cpp/dolfinx/mesh/Topology.cpp

```diff
@@ -702,28 +713,110 @@ std::vector<std::int32_t> convert_to_local_indexing(
 
 //-----------------------------------------------------------------------------
 Topology::Topology(MPI_Comm comm, CellType cell_type)
-    : _comm(comm), _cell_type(cell_type),
+    : _comm(comm), _index_map(cell_dim(cell_type) + 1, {nullptr}),
       _connectivity(
           cell_dim(cell_type) + 1,
           std::vector<std::shared_ptr<graph::AdjacencyList<std::int32_t>>>(
               cell_dim(cell_type) + 1))
 {
-  // Do nothing
+  std::int8_t tdim = cell_dim(cell_type);
+
+  // Create all the entity types in mesh, one per dimension for a single cell
+  // type mesh.
+  _entity_type_offsets.resize(tdim + 2);
+  for (std::int8_t i = 0; i < tdim + 2; ++i)
+    _entity_type_offsets[i] = i;
+
+  _entity_types = {CellType::point, CellType::interval};
```

Comment:
A bit off topic, but this excludes the option of creating point cloud meshes, as we assume that the topological dimension of the mesh is at least 1.

Reply:
That is a good point (ha ha). I will fix it. Not sure there are many applications though.